### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by BetterthanDrugs**

### CODE REVIEW

File: angular-2-components/src/app/features/registration/registration.component.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            required
+            formControlName="account_email"
+            appEmailValidate />
+          <div
+            class="alert"
+            *ngIf="
+              account_email.errors &&
+              (account_email.dirty || account_email.touched)
+            ">
+            <p *ngIf="account_email.errors?.['required']">Email is required</p>
```
Comment (MishkanJigan) [inline angular-2-components/src/app/features/registration/registration.component.html:41]: Try to use shared string `enum `for all of this string such as `Email is required` and etcetera.

File: angular-2-components/src/app/features/registration/registration.component.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  get account_email(): any {
+    return this.formDataRegistration.get('account_email');
+  }
+
+  get account_password(): any {
+    return this.formDataRegistration.get('account_password');
+  }
+
+  onSubmit() {
```
Comment (MishkanJigan) [inline angular-2-components/src/app/features/registration/registration.component.ts:45]: Don't forget about type for all functions an variables.

File: angular-2-components/src/app/features/styles/input.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+.password-icon {
+  position: absolute;
+  right: 5px;
+  top: 50%;
+  transform: translateY(-30%);
+  cursor: pointer;
+}
+
+.visually-hidden {
+  position: absolute !important;
```
Comment (MishkanJigan) [inline angular-2-components/src/app/features/styles/input.scss:53]: Is there any other way without using `important`?

File: angular-2-components/src/app/features/registration/registration.component.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@import '../styles/input.scss';
```
Comment (MishkanJigan) [inline angular-2-components/src/app/features/registration/registration.component.scss:1]: General question: why all inputs have a red border by default?

File: angular-2-components/src/app/shared/directive/email-validator.directive.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { Directive } from '@angular/core';
+import {
+  AbstractControl,
+  NG_VALIDATORS,
+  ValidationErrors,
+  Validator,
+  ValidatorFn,
+} from '@angular/forms';
+
+export const REG_EXP_EMAIL =
```
[the remark this answers is not in the document]
Comment (BetterthanDrugs) [reply angular-2-components/src/app/shared/directive/email-validator.directive.ts:10]: in the future we can reuse it

File: angular-2-components/src/app/shared/directive/email-validator.directive.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  /^(|(([A-Za-z0-9]+_+)|([A-Za-z0-9]+\-+)|([A-Za-z0-9]+\.+)|([A-Za-z0-9]+\++))*[A-Za-z0-9]+@((\w+\-+)|(\w+\.))*\w{1,63}\.[a-zA-Z]{2,6})$/i;
+
+export function emailValidator(): ValidatorFn {
```
[the remark this answers is not in the document]
Comment (BetterthanDrugs) [reply angular-2-components/src/app/shared/directive/email-validator.directive.ts:13]: in the future we can reuse it

File: angular-2-components/src/app/features/styles/input.scss (hunk 3 above)
Comment (BetterthanDrugs) [reply angular-2-components/src/app/features/styles/input.scss:53]: i will try to increase the specificity of styles

**Pull request merged**